Fix linter errors for staging/src/k8s.io/cloud-provider.

russtaylor opened the pull request:
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Cleanup of linter issues. Fixes a part of https://github.com/kubernetes/kubernetes/issues/68026.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

tossmilestone commented:
/ok-to-test

fejta commented:
/approve
/lgtm

fejta commented:
/lgtm cancel
Please pass tests

tossmilestone commented:
@russtaylor You should squash some commits together to make the PR clean, before it is merged. Thanks for you contribution!

russtaylor commented:
I've cleaned it up as much as I was able to with an interactive rebase. Let me know if this is enough, but otherwise I can redo it as a single commit.

russtaylor commented:
/retest

russtaylor commented:
/retest

russtaylor commented:
/retest

tossmilestone commented:
@russtaylor I think a single commit is preferred.

russtaylor committed:
Fix linter errors for staging/src/k8s.io/cloud-provider

The commits have been squashed together for easy merging, which required
rebuilding the branch. I hope that doesn't cause problems.

russtaylor commented:
/retest

fejta commented:
Please rebase! Can LGTM soon thereafter

russtaylor commented:
Rebase complete!

russtaylor commented:
Apparently incorrectly... I'll get that fixed...

russtaylor commented:
/retest

russtaylor commented:
/retest

fejta commented:
Thanks!
/lgtm

fejta commented:
Will you please /approve?

For cloudprovider
/assign @mikedanese @dims 

For azure: 
/assign @saad-ali @msau42 

etc

dims commented:
/assign @andrewsykim 
/unassign

andrewsykim commented on pkg/cloudprovider/providers/fake/fake.go:
```diff
@@ -26,7 +26,7 @@ import (
 
 	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
-	cloudprovider "k8s.io/cloud-provider"
```
This would result in compile error below with `cloudprovider.ErrNotImplemented`.

andrewsykim replied:
wow, TIL this still compiles. I prefer we keep it still for consistency, what do you think? :)

andrewsykim commented on pkg/cloudprovider/providers/gce/gce_disks.go:
```diff
@@ -28,7 +28,7 @@ import (
 	"k8s.io/apimachinery/pkg/api/resource"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/sets"
-	cloudprovider "k8s.io/cloud-provider"
```
same here, I think we want to keep the `cloudprovider` alias here.

andrewsykim commented on pkg/cloudprovider/providers/gce/gce_disks_test.go:
```diff
@@ -26,7 +26,7 @@ import (
 	compute "google.golang.org/api/compute/v1"
 	"google.golang.org/api/googleapi"
 	"k8s.io/apimachinery/pkg/util/sets"
-	cloudprovider "k8s.io/cloud-provider"
```
same here

andrewsykim commented on pkg/cloudprovider/providers/gce/gce_instances.go:
```diff
@@ -33,7 +33,7 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/wait"
-	cloudprovider "k8s.io/cloud-provider"
```
same here

andrewsykim commented on pkg/cloudprovider/providers/ovirt/ovirt.go:
```diff
@@ -33,7 +33,7 @@ import (
 
 	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
-	cloudprovider "k8s.io/cloud-provider"
```
same here

andrewsykim commented on pkg/cloudprovider/providers/vsphere/vsphere.go:
```diff
@@ -40,7 +40,7 @@ import (
 	k8stypes "k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/informers"
 	"k8s.io/client-go/tools/cache"
-	cloudprovider "k8s.io/cloud-provider"
```
same here :)

andrewsykim commented on staging/src/k8s.io/cloud-provider/cloud.go:
```diff
@@ -74,11 +75,12 @@ type Clusters interface {
+// LoadBalancers that were created prior to Kubernetes v1.12.
+//
+// Deprecated: Each provider should replace this method call in GetLoadBalancerName with a
+// provider-specific implementation is less cryptic than the Service's UUID.
```
`implementation is less cryptic ` -> `implementation that is less cryptic `

andrewsykim commented:
Thanks @russtaylor! PR looks great overall. Changing the err names will "break" some of the out-of-tree providers but this is fine given we let them know. We're trying to have any changes to `k8s.io/cloud-provider` pass through SIG Cloud Provider for approval (even trivial changes like this one) given the list of providers is growing and changes there may have implications we have not considered. I've added this PR to next meeting's agenda, thanks! cc @cheftako

msau42 commented:
> For azure:
> /assign Saad Ali Michelle Au

I think @feiskyer may be better for azure?

I took a look at gce_disk and lgtm

feiskyer commented on pkg/cloudprovider/providers/azure/azure_wrap.go:
```diff
@@ -81,7 +81,7 @@ func (az *Cloud) getVirtualMachine(nodeName types.NodeName) (vm compute.VirtualM
 
 	if cachedVM == nil {
-		return vm, cloudprovider.InstanceNotFound
+		return vm, cloudprovider.ErrInstanceNotFound
```
azure part lgtm
